syntax = "proto3";

import "google/protobuf/timestamp.proto";

service EnergyDataService {
    rpc GetSensorData(GetSensorDataRequest) returns (stream GetSensorDataResponse) {}
    rpc GetStats(GetStatsRequest) returns (GetStatsResponse) {}
}

message DataPoint {
    google.protobuf.Timestamp timestamp = 1;
    float energy_consumed = 2;
    float active_power = 3;
    float voltage = 4;
    float current = 5;
    float factor = 6;
    float frequency = 7;
}

message GetSensorDataRequest {
    string consumer_id = 1;
}

message GetSensorDataResponse {
    repeated DataPoint sensor_data = 1;
}

message GetStatsRequest {
    string consumer_id = 1;
}

message GetStatsResponse {
    float yesturdays_usage = 1;
    float todays_usage = 2;
    float delta_usage = 3;
    float current_bill = 4;
    float projected_bill = 5;
}
